### PR TITLE
Set js-parser feature in the build script.

### PR DESCRIPTION
**Pull request by robertsipka**
This patch also contains:
- Renamed `FEATURE_PARSER_DISABLE` to `FEATURE_JS_PARSER`.
- Fixed the build error that occurred in case of disabled js-parser.

### CODE REVIEW

**Comment by galpeter** [comment]
lgtm (I like the `FEATURE_JS_PARSER` name:))

**Comment by zherczeg** [comment]
There is one thing I am worry about. If somebody just make an all-in-one build, and forget to explicitly define this variable, he will not get a parser and will be surprised. Due to a similar reason we changed float32 default to float64 default. I think we should check somewhere that either snapshot or parser must be enabled, and use # error to signal when both are disabled. If somebody uses the python build system, this could never happen, but if somebody just takes all sources and tries to compile, it may.

**Comment by robertsipka** [comment]
@zherczeg : I agree with you. I've added an #error to signal when both are disabled.

File: jerry-core/jerry.c [hunk 1]
```diff
@@ -49,6 +49,10 @@ JERRY_STATIC_ASSERT ((int) ECMA_ERROR_COMMON == (int) JERRY_ERROR_COMMON
                      && (int) ECMA_ERROR_URI == (int) JERRY_ERROR_URI,
                      ecma_standard_error_t_must_be_equal_to_jerry_error_t);
 
+#if !defined (JERRY_JS_PARSER) && !defined (JERRY_ENABLE_SNAPSHOT_EXEC)
+#error JS_PARSER or JERRY_ENABLE_SNAPSHOT_EXEC must be defined!
+#endif
```
Comment (zherczeg) [inline jerry-core/jerry.c]: Comment missing after endif.

File: jerry-core/jerry.c [hunk 2]
```diff
@@ -49,6 +49,10 @@ JERRY_STATIC_ASSERT ((int) ECMA_ERROR_COMMON == (int) JERRY_ERROR_COMMON
                      && (int) ECMA_ERROR_URI == (int) JERRY_ERROR_URI,
                      ecma_standard_error_t_must_be_equal_to_jerry_error_t);
 
+#if !defined (JERRY_JS_PARSER) && !defined (JERRY_ENABLE_SNAPSHOT_EXEC)
+#error JS_PARSER or JERRY_ENABLE_SNAPSHOT_EXEC must be defined!
```
Comment (galpeter) [inline jerry-core/jerry.c]: Should `JS_PARSER` be `JERRY_JS_PARSER` ?

**Comment by robertsipka** [comment]
@zherczeg , @galpeter : Thanks, I've added these minor fixes.

**Pull request merged**